### PR TITLE
Flash: Add multi-key management support

### LINKED ISSUE #1999: Flash: Add multi-user management support
After the discussion with the Devops team we came to a conclusion that the current implementation which uses an instance of a flash node for each keypair is not really scalable. The mobile/web wallets currently use a single Agora/Stoa node.

It should be possible to have a single `FlashAPI` node responsible for many keypairs.

@MichaelKim20 @TrustHenry I will keep you updated on the progress of this issue.

### PR DESCRIPTION
**Pull request by AndrejMitrovic**
Fixes #1999

@TrustHenry @MichaelKim20 this will allow a single Flash node to manage multiple key-pairs. I will work on the rest of the issues now. 

In the current design the key-pairs are still shared with the Flash node. It might be possible in the future to change this so that the Wallet is doing signing and sharing the signature with the Flash node. I'll file a separate issue for that.

### CODE REVIEW

**Comment by AndrejMitrovic** [comment]
> faucet: Unable to retrieve function name

~I think this has something to do with the function signatures changing but faucet still using the old Agora node. Although I'm not sure where this string originates from (maybe vibe.d?). Only the Flash APIs were changed though.~

~We might need to do the merge + do a quick PR to update the agora submodule in faucet after.~

Edit: Never mind, had to rebase.

@omerfirmak if you don't have any comments I'll get this going soon.

**Comment by AndrejMitrovic** [comment]
Windows issues:

0x00007FFB09899689 (0x00006CCED9DC2DF6 0x00007FFB0D7B803D 0x0000000000000008 0x00007FF766434F9F)
0x00007FF7663F0A18 (0x00000199CEF7AF58 0x0000019996118FD8 0x0000000000018000 0x0000019996118FD8)
0x00007FF7663ECD7B (0x0000000000000000 0x0000006800000000 0x0000019996118F70 0x00000199A05FBDF0)
0x00007FF7663E8E9D (0x00000199D2AF0750 0x0000019900000000 0x00006CCED9DC2D96 0x00007FF7BF3BC900)
0x00007FF765391C06 (0x0000019996118FD8 0x00007FF764276539 0x00000068B6BFD390 0x00000068B6BFD260)
0x00007FF765391B5D (0x00000199CED5EFB8 0x00000199CEF7AF58 0x00000068B6BFD220 0x00000068B6BFD260)
0x00007FF7655549B5 (0x0000DE938FE00503 0x00000199D04D86F0 0x0000000000000031 0x0000000000070302)
0x00007FF765D92B5E (0x0000019910AD9148 0x00007FF766434BE8 0x0000000000000000 0x00000000000000E8)
0x00007FF765B781F9 (0x00000199B1EAF830 0x00000199AD2BA020 0x00000199BF3BC918 0x00007FF765B7A504)
0x00007FF765B71EB6 (0x0000019910AD9148 0x000001996D144010 0x000001998C750560 0x00000068B6BFD6E0)
0x00007FF765B7A2C4 (0x00000199957258A8 0x0000006800000000 0x00000199961B29B0 0x0000000000000200)
0x00007FF7650F9A2D (0x0000DE938FE08F8E 0x0000000000000001 0x0000000000000003 0x0000019910AD9248)
0x00007FF764269221 (0x0000000000000000 0x0000000000000001 0x00000068B6BFDBF0 0x000001980DC00DD0)
0x00007FF764263E22 (0x005F00630064006C 0x0032002E00000000 0x002D0030002E0035 0x0035003100450039)
0x00007FF7663901C1 (0x000001998F5FC010 0x000001980BF40000 0x000001980BDF0F60 0x0000000000008000)
0x00007FF76638DE46 (0x00000068B6BFE678 0x00000068B6BFE678 0x0000000000000017 0x000001980DC270B0)
0x00007FF766361AD7 (0x00007FF7676C3170 0x0000000000000000 0x000001980BDF0F60 0x0000000000008000)
0x00007FF7662CF407 (0x0000000800000023 0x000001980BF40320 0x000000000000000D 0x000001980DA1ED90)
0x00007FF766265531 (0x0000000000000000 0x0000000000000000 0x0000000000000000 0x00007FFB0D7A99E0)
0x00007FF7662CBCDF (0x0000019806000006 0x006900740000002F 0x0000000000000020 0x0000000000000004)
0x00007FF7663CEB69 (0x0000000000000001 0x00007FFB098B7456 0x00007FF76774B383 0x0000000000000000)
0x00007FF7663CE7FD (0x000001980BF55958 0x0000000000000000 0x00000068B6BFFCA0 0x0000000000000001)
0x00007FF7663CEAF4 (0x000001980BF55962 0x0000000000000059 0x000001980BF55962 0x00007FF700000000)
0x00007FF7662C7719 (0x0000000000000000 0x0000000000000000 0x0000000000000000 0x0000000000000000)
0x00007FF766413D78 (0x0000000000000000 0x0000000000000000 0x0000000000000000 0x0000000000000000)
0x00007FFB0D6C7974 (0x0000000000000000 0x0000000000000000 0x0000000000000000 0x0000000000000000)
0x00007FFB0D7FA2F1 (0x0000000000000000 0x0000000000000000 0x0000000000000000 0x0000000000000000)
ldc2 failed with exit code -529697949.

**Comment by Geod24** [comment]
https://github.com/ldc-developers/ldc/issues/3702

**Review by omerfirmak — APPROVED** [review]
I think we could probably get away with not adding the PublicKey to some of the APIs but it's OK. We could go back to it if we ever feel like it:)

**Comment by AndrejMitrovic** [comment]
> I think we could probably get away with not adding the PublicKey to some of the APIs but it's OK. We could go back to it if we ever feel like it:)

Yes I totally agree. Thanks for the approval.

**Comment by AndrejMitrovic** [comment]
Is there any workaround we can use for the Windows build? I guess temporarily disabling it would be going too far.

**Comment by AndrejMitrovic** [comment]
I added another `exclude` line for the last ldc release + windows. Let's see if this works.

**Comment by AndrejMitrovic** [comment]
Huh why is the Windows build still pending? Did I do it wrong?

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -33,6 +33,7 @@ jobs:
         exclude:
```
Comment (TrustHenry) [inline .github/workflows/main.yml]: exclude test a `dc: ldc-1.25.0, os: windows-2019`

Comment (AndrejMitrovic) [reply .github/workflows/main.yml]: Sorry? I don't follow.

Comment (TrustHenry) [reply .github/workflows/main.yml]: os: [ ubuntu-18.04, macOS-10.15 ]  just run :)

Comment (AndrejMitrovic) [reply .github/workflows/main.yml]: Ahh you mean to not use exclude. Right.

Comment (AndrejMitrovic) [reply .github/workflows/main.yml]: Thank you @TrustHenry the Wise.

**Pull request merged**